monitoring/src/main/java/org/polypheny/db/monitoring/statistics/StatisticsManagerImpl.java

```diff
@@ -806,86 +879,113 @@ public void tablesToUpdate( Long tableId ) {
+        if ( schema != null ) {
+            Map<Long, StatisticColumn<T>> table = this.statisticSchemaMap.get( schemaId ).get( tableId );
+            if ( table != null ) {
+                table.remove( changedValues.keySet().stream().findFirst().get() );
+            }
+        }
+    }
+
+
+    private void handleTruncate( Long tableId, Long schemaId, Catalog catalog ) {
+        CatalogTable catalogTable = catalog.getTable( tableId );
+        for ( int i = 0; i < catalogTable.columnIds.size(); i++ ) {
+            PolyType polyType = catalog.getColumn( catalogTable.columnIds.get( i ) ).type;
+            QueryColumn queryColumn = new QueryColumn( schemaId, catalogTable.id, catalogTable.columnIds.get( i ), polyType );
+            if ( this.statisticSchemaMap.get( schemaId ).get( tableId ).get( catalogTable.columnIds.get( i ) ) != null ) {
+                StatisticColumn statisticColumn = createNewStatisticColumns( polyType, queryColumn );
+                if ( statisticColumn != null ) {
+                    put( queryColumn, statisticColumn );
                 }
-            } else {
-                addDataStatistics( changedValues, catalog, catalogTable, columns );
             }
-        } else if ( type.equals( "TRUNCATE" ) ) {
-            Catalog catalog = Catalog.getInstance();
-            if ( catalog.checkIfExistsTable( tableId ) ) {
-                CatalogTable catalogTable = catalog.getTable( tableId );
-                for ( int i = 0; i < catalogTable.columnIds.size(); i++ ) {
-                    PolyType polyType = catalog.getColumn( catalogTable.columnIds.get( i ) ).type;
-                    QueryColumn queryColumn = new QueryColumn( schemaId, catalogTable.id, catalogTable.columnIds.get( i ), polyType );
-                    if ( this.statisticSchemaMap.get( schemaId ).get( tableId ).get( catalogTable.columnIds.get( i ) ) != null ) {
-                        if ( polyType.getFamily() == PolyTypeFamily.NUMERIC ) {
-                            NumericalStatisticColumn numericalStatisticColumn = new NumericalStatisticColumn<>( queryColumn );
-                            put( queryColumn, numericalStatisticColumn );
-                        } else if ( polyType.getFamily() == PolyTypeFamily.CHARACTER ) {
-                            AlphabeticStatisticColumn alphabeticStatisticColumn = new AlphabeticStatisticColumn<T>( queryColumn );
-                            put( queryColumn, alphabeticStatisticColumn );
-                        } else if ( PolyType.DATETIME_TYPES.contains( polyType ) ) {
-                            TemporalStatisticColumn temporalStatisticColumn = new TemporalStatisticColumn<T>( queryColumn );
-                            put( queryColumn, temporalStatisticColumn );
-                        }
+        }
+    }
+
+
+    private StatisticColumn createNewStatisticColumns( PolyType polyType, QueryColumn queryColumn ) {
```

Comment:
Raw use of class StatisticColumn (no only here but also at several other places)

Reply:
Fixed in commit: 899dfe4